nucliadb_node/src/grpc/grpc_reader.rs

```diff
@@ -32,14 +32,14 @@ use crate::shards::providers::AsyncShardReaderProvider;
 use crate::shards::reader::ShardReader;
 
 pub struct NodeReaderGRPCDriver {
-    shards: AsyncUnboundedShardReaderCache,
+    pub shards: Arc<AsyncUnboundedShardReaderCache>,
```

Comment:
It's a bad practice to publish fields like this, as you are creating dependencies between layers of abstraction knowing too many implementation details.

The cache provider can be passed to `NodeReaderGRPCDriver`'s constructor, so you don't need to expose internal fields.